### PR TITLE
Blog post about critical css

### PR DESCRIPTION
**Pull request by shidhincr**

### CODE REVIEW

**Comment by shidhincr** [comment]
@razanbilwani Can you review .. :)

File: source/_posts/2015-04-08-improve-website-page-load-performance-with-critical-css.markdown [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+date: 2015-04-08 16:41
+comments: true
+author: Shidhin CR
+categories: 
+- critical css
+- nginx
+- nodejs
+---
+
+![Critical Css](/images/critical-css.png)
```
Comment (geshan) [inline source/_posts/2015-04-08-improve-website-page-load-performance-with-critical-css.markdown]: where is the image? is it already there?

File: source/_posts/2015-04-08-improve-website-page-load-performance-with-critical-css.markdown [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+## FOUC  ( Flash of Unstyled Content )
+
+**FOUC** is the abbreviation for Flash of Unstyled Content.  A **FOUC** can occur when the browser has already parsed the DOM, then apply the stylesheet later.  An user at this time will see a completely un-styled content till the stylesheet is downloaded and applied. This is the reason for including stylesheets in the `HEAD` tag considered to be a best practice. Because, if the resources are in the `HEAD`, browser has to download and apply all of them before parsing the DOM. 
+
+Including the stylesheets in the `HEAD` is a best practice for good user experience. But, it hit the page load performance in the opposite direction. Since the browser has to download all the resources in the `HEAD` tag before any parsing, the page load time increases. This particular problem existed for years without any solution, and now, we have something known as **Critical Css** to fix it.
```
Comment (geshan) [inline source/_posts/2015-04-08-improve-website-page-load-performance-with-critical-css.markdown]: `it hits the`

File: source/_posts/2015-04-08-improve-website-page-load-performance-with-critical-css.markdown (hunk 1 above)
Comment (shidhincr) [inline source/_posts/2015-04-08-improve-website-page-load-performance-with-critical-css.markdown]: Yeah.. it's there, but won't be visible if you use Github preview

**Comment by odino** [comment]
:+1: 

@shidhincr let me know when we should publish :)

File: source/_posts/2015-04-08-improve-website-page-load-performance-with-critical-css.markdown [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+It should be fairly easy to configure the pagespeed module. However, if you are using a CDN for serving your assets, you might need to add these filters in your configuration:
+
+```sh
+pagespeed InlineResourcesWithoutExplicitAuthorization
+pagespeed MapProxyDomain
+```
+
+If you use Docker, we have something for you. Here in Namshi, we open-sourced a docker container running nginx with pagespeed module. Check out this link and read the documentation:
+
+[Docker Node Nginx Pagespeed](https://github.com/namshi/docker-node-nginx-pagespeed)
```
Comment (unlucio) [inline source/_posts/2015-04-08-improve-website-page-load-performance-with-critical-css.markdown]: the container its self comes form a nodejs container, perhaps add "If you use Docker and node.js"?

**Comment by unlucio** [comment]
:+1:  mega
btw, since you're at it: wanna publish [this guy](https://github.com/namshi/misc/tree/master/node/criticalCssModule) as well?

File: source/_posts/2015-04-08-improve-website-page-load-performance-with-critical-css.markdown (hunk 3 above)
Comment (shidhincr) [inline source/_posts/2015-04-08-improve-website-page-load-performance-with-critical-css.markdown]: :+1:

**Comment by shidhincr** [comment]
Published .. Merging now

**Pull request merged**